lib/GenomeFileUtil/GenomeFileUtilImpl.py

```diff
@@ -895,6 +1062,113 @@ def save_one_genome(self, ctx, params):
+        # input should only be of workspace object types:
+        #   - 'KBaseGenomes.Genome'
+        #   - 'KBaseGenomeAnnotations.Assembly'
+        #   - 'KBaseGenomes.ContigSet'
+        obj_type = ws.get_object_info3({'objects': [{'ref': input_ref}]})['infos'][0][2]
+        if ("KBaseGenomes.Genome" not in obj_type 
+          and "KBaseGenomes.ContigSet" not in obj_type 
+          and "KBaseGenomeAnnotations.Assembly" not in obj_type
+        ):
+          raise ValueError(f"input reference must be of type KBaseGenomes.Genome or "
+                           f"KBaseGenomes.ContigSet or KBaseGenomeAnnotations.Assembly "
+                           f"not {obj_type}")
+
+        fasta_file = [val['paths'][0] for key, val in au.get_fastas({'ref_lst': [input_ref]}).items()][0]
+
+        params['fasta_file'] = {
+          'path': fasta_file
+        }
+        params['use_existing_assembly'] = input_ref
+        importer = FastaGFFToGenome(self.cfg)
+        returnVal = importer.import_file(params)
+
+        #END ws_obj_gff_to_genome
+
+        # At some point might do deeper type checking...
+        if not isinstance(returnVal, dict):
+            raise ValueError('Method ws_obj_gff_to_genome return value ' +
+                             'returnVal is not type dict as required.')
+        # return the results
+        return [returnVal]
+
+    def ws_obj_gff_to_metagenome(self, ctx, params):
+        """
+        :param params: instance of type "WsObjGFFToMetagenomeParams" ->
+           structure: parameter "ws_ref" of String, parameter "gff_file" of
+           type "File" -> structure: parameter "path" of String, parameter
+           "shock_id" of String, parameter "ftp_url" of String, parameter
+           "genome_name" of String, parameter "workspace_name" of String,
+           parameter "source" of String, parameter "metadata" of type
+           "usermeta" -> mapping from String to String, parameter
+           "generate_missing_genes" of type "boolean" (A boolean - 0 for
+           false, 1 for true. @range (0, 1))
+        :returns: instance of type "MetagenomeSaveResult" -> structure:
+           parameter "metagenome_ref" of String
+        """
+        # ctx is the context object
+        # return variables are: returnVal
+        #BEGIN ws_obj_gff_to_metagenome
+        au = AssemblyUtil(self.cfg.callbackURL)
+        ws = Workspace(url=self.cfg.workspaceURL)
+
+        if params.get('ws_ref'):
+          input_ref = params['ws_ref']
+        else:
+          raise ValueError(f"ws_ref argument required for ws_obj_gff_to_metagenome method")
+        # input should only be of workspace object types:
+        #   - 'KBaseMetagenomes.AnnotatedMetagenomeAssembly'
+        #   - 'KBaseGenomeAnnotations.Assembly'
+        #   - 'KBaseGenomes.ContigSet'
+        obj_type = ws.get_object_info3({'objects': [{'ref': input_ref}]})['infos'][0][2]
+        if ("KBaseMetagenomes.AnnotatedMetagenomeAssembly" not in obj_type 
```

Comment:
I would also do the `if obj_type not in valid_types` syntax here to be more readable